aggregate_spatial.json

```diff
@@ -0,0 +1,80 @@
+{
+    "id": "aggregate_spatial",
+    "summary": "Zonal statistics for geometries",
+    "description": "Aggregates statistics for one or more geometries (e.g. zonal statistics for polygons) over the spatial dimensions. This process passes a list of values to the reducer. In contrast, aggregate_spatial_binary() passes two values, which may be better suited for UDFs.\n\n- For **polygons**, the process considers all pixels for which the point at the pixel center intersects with the corresponding polygon (as defined in the Simple Features standard by the OGC).\n- For **points** and **lines** (line strings), the process considers all the closest pixel centers.\n\nThus, pixels may be part of multiple geometries and be part of multiple aggregations.\n\nThe data cube must have been reduced to only contain two spatial dimensions and a third dimension the values are aggregated for, for example the temporal dimension to get a time series. Otherwise this process fails with the `TooManyDimensions` error.\n\nThe number of total and valid pixels is returned together with the calculated values.",
```

Comment:
I'm still unsure about the specification for "closest pixel centres" for a line. Closest to what? A line is made of infinite points, and pixel centres will be closer or further away from the line depending on the angle. For me, only a buffer of half the pixel size away from the line makes sense. Of course, if the backend developers know what "closest pixels to a line" means, then that's also fine for me as it is.

Reply:
I was also confused by that, but was proposed as it is in one of the calls. Maybe someone can clarify? Maybe @jdries or @mkadunc ?

Reply:
Closest to at least one out of the infinite points on the line — if the raster grid is square, or if "raster cell" is the voronoi cell of the raster points, then:
- "process considers all the pixels whose centers are closest to at least one point on the line"

 is equivalent to saying
- "process considers all the pixels whose cells intersect the geometry"

...but without mixing up pixel center-points and pixel cells in the same paragraph and going into the whole raster-is-area vs. raster-is-point philosophical debate.

Reply:
The second point is clear to me. It's fine if all backends understand it that way, then.

Reply:
@mkadunc Thanks, I updated the process definitions accordingly:

- For **polygons**, the process considers all pixels for which the point at the pixel center intersects with the corresponding polygon (as defined in the Simple Features standard by the OGC).
- For **points**, the process considers the closest pixel center.
- For **lines** (line strings), the process considers all the pixels whose centers are closest to at least one point on the line.